packages/schema/src/types/module.ts

```diff
@@ -73,6 +80,9 @@ export interface ModuleDefinition<
   defaults?: TOptionsDefaults | ((nuxt: Nuxt) => Awaitable<TOptionsDefaults>)
   schema?: TOptions
   hooks?: Partial<NuxtHooks>
+  // TODO: review option name
+  // TODO: type constraints for module options
+  modules?: Record<string, ModuleDependencyMeta>
```

Comment:
let's review this - we might also be able to do something with `modules.d.ts` to ensure the types match ....